install.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+# This is an install script for notes
+
+function assertInstalled() {
+    for var in "$@"; do
+        if ! which $var &> /dev/null; then
+            echo "$var is required but not installed, exiting."
+            exit 1
+        fi
+    done
+}
+
+# Yay, Echo self documents! :D
+echo "Checking for root..."
+    [ "$(whoami)" != "root" ] && exec sudo -- "$0" "$@"
+
+# Make sure we have everything actually installed that we need to install this.
+echo "Checking for Dependencies..."
+    assertInstalled bash curl tar mktemp install make
```

Comment:
I'm not sure if we could do that check for install inside Makefile, since making functions inside Makefiles is super hard. and I think that having one assertInstalled is better than two.